File: src/main/java/people/male/Man.java
package people.male;

import com.google.inject.Inject;

public class Man implements IMan {

    private String firstName;
    private String lastName;

    @Inject
    public Man(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public String getFullName() {
        return firstName + " " + lastName;
    }

    @Override
    public String getFirstName() {
        return firstName;
    }

    @Override
    public String getLastName() {
        return lastName;
    }

    @Override
    public IMan setFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }

    @Override
    public IMan setLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }
}

File: src/kotlin/KotlinExamples/src/Man.kt
import com.google.inject.Inject

class Man @Inject
constructor(override var firstName: String?, override var lastName: String?) : IMan {

    override val fullName: String? get() = "$firstName $lastName"

    override fun setFirstName(firstName: String): IPerson {
        this.firstName = firstName
        return this
    }

    override fun setLastName(lastName: String): IPerson {
        this.lastName = lastName
        return this
    }
}